termux-exec.c

```diff
@@ -231,8 +236,33 @@ int execve(const char* filename, char* const* argv, char* const* envp)
 		}
 	}
 
-	int ret = real_execve(filename, argv, envp);
+	int ret = real_exec_fn(filename, argv, envp);
 	free(new_argv);
 	free(new_envp);
 	return ret;
+	
+}
+
+int _execvp(const char* filename, char* const* argv, char* const* envp)
+{
+        (void)envp;
+        int (*real_execvp) (const char*, char* const[]) = dlsym(RTLD_NEXT, "execvp");
+        return real_execvp(filename, argv);;
+}
+
+int _execve(const char* filename, char* const* argv, char* const* envp)
+{
+        int (*real_execve)(const char*, char* const[], char* const[]) = dlsym(RTLD_NEXT, "execve");
+        return real_execve(filename, argv, envp);
+}
+
+int execve(const char* filename, char* const* argv, char* const* envp)
+{
+        return exec_wrapper(filename, argv, envp, _execve);
+}
+
+int execvp(const char* filename, char* const argv[])
+{
+        char* null_envp[] = { NULL };
```

Comment:
This looks wrong. According to `man execvp`:

>All other exec() functions (which do not include 'e' in the suffix) take the environment for the new process image from the external variable environ in the calling process.

Here, "all other exec() functions" do include `excvp` here.

Reply:
However, according to the function definition, execvp() does not take environment variables, but another function, execvpe, requires envp. 
```
int execvp(const char *file, char *const argv[]);
int execvpe(const char *file, char *const argv[], char *const envp[]);
```

Reply:
<img width="1040" alt="Screenshot 2024-02-27 at 3 43 32 PM" src="https://github.com/qiongwu1016/termux-exec/assets/70813818/f51e4c66-6c5a-4dd9-b052-2e84d58b68d5">